cypress/e2e/tests/pages/user_menu/preferences.spec.ts

```diff
@@ -0,0 +1,368 @@
+        prefPage.listBox().isOpened();
+      }
+    });
+  });
+
+  it('Can select time format', () => {
+    /*
+    Select each option
+    Get values of options available and compare them to Regex
+    */
+    const dropBoxIndex = 3;
+
+    PagePo.goTo(endpoint);
+    prefPage.dropdownMenu().open(dropBoxIndex);
+    prefPage.listBox().isOpened();
+    prefPage.listBox().getListBoxItems().should('have.length', 2).then(($els) => {
+      const map = Cypress.$.map($els, el => el.innerText.trim());
+
+      for (const i in map) {
+        expect(map[i]).to.match(/([0-9]+(:[0-9]+)+)/);
+        prefPage.listBox().set(map[i]);
+        prefPage.listBox().isClosed();
+        prefPage.dropdownMenu().open(dropBoxIndex);
+        prefPage.listBox().isOpened();
+      }
+    });
+  });
+
+  it('Can select Table Rows per Page', () => {
+    /*
+    Select each option
+    Get values of options available and compare them to 'options' list
+    */
+    const dropBoxIndex = 4;
+    const options = ['10', '25', '50', '100'];
+
+    PagePo.goTo(endpoint);
+    for (const i in options) {
+      prefPage.dropdownMenu().open(dropBoxIndex);
+      prefPage.listBox().isOpened();
+      prefPage.listBox().getListBoxItems().should('have.length', 4);
+      prefPage.listBox().set(options[i]);
+      prefPage.dropdownMenu().checkOptionSelected(dropBoxIndex, options[i]).then((selectedOption) => {
+        const map = Cypress.$.map(selectedOption, text => text.innerText.trim());
+
+        for (const v in map) {
+          expect(map[v]).to.eq(options[i]);
+        }
+      });
+      prefPage.listBox().isClosed();
+    }
+  });
+
+  it('Can select Number of clusters to show in side menu ', () => {
+    /*
+    Select each option
+    Get values of options available and compare them to 'options' list
+    */
+    const dropBoxIndex = 5;
+    const options = ['2', '3', '4', '5', '6', '7', '8', '9', '10'];
+
+    PagePo.goTo(endpoint);
+    for (const i in options) {
+      prefPage.dropdownMenu().open(dropBoxIndex);
+      prefPage.listBox().isOpened();
+      prefPage.listBox().getListBoxItems().should('have.length', 9);
+      prefPage.listBox().set(options[i]);
+      prefPage.dropdownMenu().checkOptionSelected(dropBoxIndex, options[i]).then((selectedOption) => {
+        const map = Cypress.$.map(selectedOption, text => text.innerText.trim());
+
+        for (const v in map) {
+          expect(map[v]).to.eq(options[i]);
+        }
+      });
+      prefPage.listBox().isClosed();
+    }
+  });
+
+  it('Can select Confirmation Setting', () => {
+    /*
+    Select the checkbox and verify state is preserved after logout/login
+    */
+    const checkboxLabel = 'Do not ask for confirmation when scaling down node pools.';
+
+    PagePo.goTo(endpoint);
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isChecked();
+    userMenu.open();
+    userMenu.checkOpen();
+    userMenu.clickMenuLink('Log Out');
+    cy.login();
```

Comment:
having a log out / in / go to within each test is going to make this a very long running spec. can this be done for only one test? alternatively can the state after log in all be checked in a single `it` at the end?

Reply:
good call! I've removed all logout/login from all tests except one: `Can select login landing page`. That saves a lot of test execution time.